src/openrct2-ui/windows/Map.cpp

```diff
@@ -1389,17 +1384,11 @@ static void map_window_increase_map_size()
  */
 static void map_window_decrease_map_size()
 {
-    if (gMapSize < 16)
-    {
-        context_show_error(STR_CANT_DECREASE_MAP_SIZE_ANY_FURTHER, STR_NONE, {});
+    auto decreaseMapSizeAction = ChangeMapSizeAction(-1);
+    auto queryResult = GameActions::Query(&decreaseMapSizeAction);
+    if (queryResult->Error != GameActions::Status::Ok)
```

Comment:
Ummm what? Calling GameActions::Execute calls a Query already and only if the query fails then it does the execute. What you have written now performs the query twice then the execute.